### PR TITLE
Add curl_global_init/cleanup to NetworkCurl

### PR DESCRIPTION
**Pull request by meravingen123**
* Added curl_global_init() and curl_global_cleanup()
  function calls to linux network implementation which
  uses curl to do proper initialization/deinitialization
  in network.

Resolves: OLPEDGE-1536

Signed-off-by: Serhii Lozynskyi <ext-serhii.lozynskyi@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/include/olp/core/client/OlpClientSettingsFactory.h [hunk 1]
```diff
@@ -58,6 +58,13 @@ class CORE_API OlpClientSettingsFactory final {
    *
    * Defaulted to platform-specific implementation.
    *
+   * @note Network implementation on linux uses curl library. Calling this
+   * function will call curl_global_init() function which should be called on
+   * the start of application and cannot be called from static variable
+   * initialization and when there is more than one thread running. Please take
+   * a look at curl documentation:
+   * https://curl.haxx.se/libcurl/c/curl_global_init.html
```
Comment (HalynaDumych) [inline olp-cpp-sdk-core/include/olp/core/client/OlpClientSettingsFactory.h]:
```suggestion
   * @note It is a network implementation on Linux that uses the cURL library.
   * Calls the `curl_global_init()` function that should be called on
   * the start of an application and cannot be called from a static variable
   * initialization and when there is more than one thread running. 
   * 
   * @see [cURL documentation](Lhttps://curl.haxx.se/libcurl/c/curl_global_init.html) for more information.
```

File: olp-cpp-sdk-core/src/http/curl/NetworkCurl.h [hunk 2]
```diff
@@ -333,6 +333,9 @@ class NetworkCurl : public olp::http::Network,
   /// network transfer.
   std::unique_ptr<std::mutex[]> ssl_mutexes_{};
 #endif
+
+  // Used to store if curl_global_init(0 was ssuccessfull on construction
```
Comment (HalynaDumych) [inline olp-cpp-sdk-core/src/http/curl/NetworkCurl.h]: Should it be public and rendered to Doxygen? If yes, please change to:
```suggestion
  /**
   * Stores if `curl_global_init(0)` was ssuccessfull on construction.
   */
```
Please also specify what it stores (e.g. Stores data if...).

File: olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp [hunk 3]
```diff
@@ -227,6 +227,12 @@ NetworkCurl::NetworkCurl(size_t max_requests_count)
   OLP_SDK_LOG_TRACE(kLogTag, "Created NetworkCurl with address="
                                  << this
                                  << ", handles_count=" << max_requests_count);
+  auto error = curl_global_init(CURL_GLOBAL_ALL);
+  curl_initialized_ = error == CURLE_OK;
```
[the remark this answers is not in the document]
Comment (meravingen123) [reply olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp]: Later error is used in log to report user the error that happened.

File: olp-cpp-sdk-core/src/http/curl/NetworkCurl.h (hunk 2 above)
Comment (meravingen123) [reply olp-cpp-sdk-core/src/http/curl/NetworkCurl.h]: That is internal code and is not related to public headers. It is in doxygen probably but i dont think that it should be there. Anyways that is a comment more for developers of SDK.

Fixed typos.

File: olp-cpp-sdk-core/include/olp/core/client/OlpClientSettingsFactory.h [hunk 4]
```diff
@@ -58,6 +58,15 @@ class CORE_API OlpClientSettingsFactory final {
    *
    * Defaulted to platform-specific implementation.
    *
+   * @note It is a network implementation on Linux that uses the cURL library.
+   * Calls the `curl_global_init()` function that should be called on
+   * the start of an application and cannot be called from a static variable
+   * initialization and when there is more than one thread running.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/client/OlpClientSettingsFactory.h]:
```cpp
@note On UNIX platforms the default network request handler is libcurl based and has the known issue of static initialization and cleanup that needs special care.  We recommend that for this you should consider initializing this network request handler at a very early stage, preferably as global static or from the main thread, and pass it on to every created client. For this matter, it also is not recommended to create multiple network request handlers.
```

File: olp-cpp-sdk-core/src/http/curl/NetworkCurl.h [hunk 5]
```diff
@@ -333,6 +333,9 @@ class NetworkCurl : public olp::http::Network,
   /// network transfer.
   std::unique_ptr<std::mutex[]> ssl_mutexes_{};
 #endif
+
+  /// Stores if `curl_global_init()` was successfull on construction.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-core/src/http/curl/NetworkCurl.h]: Stores what?

File: olp-cpp-sdk-core/include/olp/core/client/OlpClientSettingsFactory.h [hunk 6]
```diff
@@ -58,6 +58,18 @@ class CORE_API OlpClientSettingsFactory final {
    *
    * Defaulted to platform-specific implementation.
    *
+   * @note On UNIX platforms the default network request handler is libcurl
+   * based and has the known issue of static initialization and cleanup that
+   * needs special care.  We recommend that for this you should consider
+   * initializing this network request handler at a very early stage, preferably
+   * as global static or from the main thread, and pass it on to every created
+   * client. For this matter, it also is not recommended to create multiple
+   * network request handlers.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-core/include/olp/core/client/OlpClientSettingsFactory.h]: On UNIX platforms, the default network request handler is libcurl-based and has the known issue of static initialization and cleanup that needs special care.  Therefore, we recommend initializing this network request handler at a very early stage, preferably as global static or from the main thread, and pass it on to every created client. For this matter, it is also not recommended to create multiple network request handlers.

File: olp-cpp-sdk-core/src/http/curl/NetworkCurl.h [hunk 7]
```diff
@@ -333,6 +333,9 @@ class NetworkCurl : public olp::http::Network,
   /// network transfer.
   std::unique_ptr<std::mutex[]> ssl_mutexes_{};
 #endif
+
+  /// Stores value if `curl_global_init()` was successfull on construction.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-core/src/http/curl/NetworkCurl.h]:
```suggestion
  /// Stores value if `curl_global_init()` was successful on construction.
```

**Pull request merged**